### PR TITLE
style: replace black+isort+flake8 tooling with ruff

### PR DESCRIPTION
**Pull request by nfarabullini**
Code style and format changes moving from black+isort+flake8 to ruff.

### CODE REVIEW

File: src/gt4py/eve/datamodels/core.py [hunk 1]
```diff
@@ -660,7 +660,7 @@ def get_fields(model: Union[DataModel, Type[DataModel]]) -> utils.FrozenNamespac
         >>> fields(Model)  # doctest:+ELLIPSIS
         FrozenNamespace(...name=Attribute(name='name', default=NOTHING, ...
 
-    """  # noqa: RST201  # doctest conventions confuse RST validator
+    """  # doctest conventions confuse RST validator
```
Comment (egparedes) [inline src/gt4py/eve/datamodels/core.py:663]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/datamodels/core.py [hunk 2]
```diff
@@ -694,8 +694,8 @@ def asdict(
         ...     x: int
         ...     y: int
         >>> c = C(x=1, y=2)
-        >>> assert asdict(c) == {'x': 1, 'y': 2}
-    """  # noqa: RST301  # sphinx.napoleon conventions confuse RST validator
+        >>> assert asdict(c) == {"x": 1, "y": 2}
+    """  # sphinx.napoleon conventions confuse RST validator
```
Comment (egparedes) [inline src/gt4py/eve/datamodels/core.py:698]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/datamodels/core.py [hunk 3]
```diff
@@ -805,9 +805,12 @@ def concretize(
         overwrite_definition: If ``True``, a previous definition of the class in
             the target module will be overwritten.
 
-    """  # noqa: RST301  # doctest conventions confuse RST validator
+    """  # doctest conventions confuse RST validator
```
Comment (egparedes) [inline src/gt4py/eve/datamodels/core.py:808]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/datamodels/core.py [hunk 4]
```diff
@@ -1018,7 +1021,7 @@ def _type_converter(value: Any) -> _T:
 _KNOWN_MUTABLE_TYPES: Final = (list, dict, set)
 
 
-def _make_datamodel(  # noqa: C901  # too complex but still readable and documented
+def _make_datamodel(  # too complex but still readable and documented
```
Comment (egparedes) [inline src/gt4py/eve/datamodels/core.py:1024]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/extended_typing.py [hunk 5]
```diff
@@ -180,11 +180,11 @@ def __get__(
     ) -> NonDataDescriptor[_C, _V]: ...
 
     @overload
-    def __get__(  # noqa: F811  # redefinion of unused member
+    def __get__(  # redefinion of unused member
```
Comment (egparedes) [inline src/gt4py/eve/extended_typing.py:183]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/extended_typing.py [hunk 6]
```diff
@@ -180,11 +180,11 @@ def __get__(
...
         self, _instance: _C, _owner_type: Optional[Type[_C]] = None
     ) -> _V: ...
 
-    def __get__(  # noqa: F811  # redefinion of unused member
+    def __get__(  # redefinion of unused member
```
Comment (egparedes) [inline src/gt4py/eve/extended_typing.py:187]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/extended_typing.py [hunk 7]
```diff
@@ -351,7 +351,7 @@ def extended_runtime_checkable(
 ) -> _ProtoT: ...
 
 
-def extended_runtime_checkable(  # noqa: C901  # too complex but unavoidable
+def extended_runtime_checkable(  # too complex but unavoidable
```
Comment (egparedes) [inline src/gt4py/eve/extended_typing.py:354]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/extended_typing.py [hunk 8]
```diff
@@ -697,7 +697,7 @@ class CallableKwargsInfo:
     data: Dict[str, Any]
 
 
-def infer_type(  # noqa: C901  # function is complex but well organized in independent cases
+def infer_type(  # function is complex but well organized in independent cases
```
Comment (egparedes) [inline src/gt4py/eve/extended_typing.py:700]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/type_definitions.py [hunk 9]
```diff
@@ -14,16 +14,15 @@
...
 """Definitions of useful field and general types."""
 
-
 from __future__ import annotations
 
 import abc
 import re
 import sys
-from enum import Enum as Enum, IntEnum as IntEnum  # noqa: F401  # imported but unused
+from enum import Enum as Enum, IntEnum as IntEnum  # imported but unused
```
Comment (egparedes) [inline src/gt4py/eve/type_definitions.py:22]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/type_definitions.py [hunk 10]
```diff
@@ -43,7 +42,7 @@
 _Tc = TypeVar("_Tc", covariant=True)
 
 
-class FrozenList(Tuple[_Tc, ...], metaclass=abc.ABCMeta):  # noqa: B024   # no abstract methods
+class FrozenList(Tuple[_Tc, ...], metaclass=abc.ABCMeta):  # no abstract methods
```
Comment (egparedes) [inline src/gt4py/eve/type_definitions.py:45]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/type_validation.py [hunk 11]
```diff
@@ -113,7 +112,7 @@ def __call__(
     ) -> FixedTypeValidator: ...
 
     @overload
-    def __call__(  # noqa: F811  # redefinion of unused member
+    def __call__(  # redefinion of unused member
```
Comment (egparedes) [inline src/gt4py/eve/type_validation.py:115]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/type_validation.py [hunk 12]
```diff
@@ -125,7 +124,7 @@ def __call__(  # noqa: F811  # redefinion of unused member
     ) -> Optional[FixedTypeValidator]: ...
 
     @abc.abstractmethod
-    def __call__(  # noqa: F811  # redefinion of unused member
+    def __call__(  # redefinion of unused member
```
Comment (egparedes) [inline src/gt4py/eve/type_validation.py:127]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/type_validation.py [hunk 13]
```diff
@@ -170,7 +169,7 @@ def __call__(
     ) -> FixedTypeValidator: ...
 
     @overload
-    def __call__(  # noqa: F811  # redefinion of unused member
+    def __call__(  # redefinion of unused member
```
Comment (egparedes) [inline src/gt4py/eve/type_validation.py:172]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/eve/type_validation.py [hunk 14]
```diff
@@ -181,7 +180,7 @@ def __call__(  # noqa: F811  # redefinion of unused member
         **kwargs: Any,
     ) -> Optional[FixedTypeValidator]: ...
 
-    def __call__(  # noqa: F811,C901  # redefinion of unused member / complex but well organized in cases
+    def __call__(  # redefinion of unused member / complex but well organized in cases
```
Comment (egparedes) [inline src/gt4py/eve/type_validation.py:183]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/common.py [hunk 15]
```diff
@@ -164,10 +164,10 @@ def __repr__(self) -> str:
     def __getitem__(self, index: int) -> int: ...
 
     @overload
-    def __getitem__(self, index: slice) -> UnitRange:  # noqa: F811 # redefine unused
+    def __getitem__(self, index: slice) -> UnitRange:  # redefine unused
```
Comment (egparedes) [inline src/gt4py/next/common.py:167]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/common.py [hunk 16]
```diff
@@ -164,10 +164,10 @@ def __repr__(self) -> str:
...
         ...
 
-    def __getitem__(self, index: int | slice) -> int | UnitRange:  # noqa: F811 # redefine unused
+    def __getitem__(self, index: int | slice) -> int | UnitRange:  # redefine unused
```
Comment (egparedes) [inline src/gt4py/next/common.py:170]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/common.py [hunk 17]
```diff
@@ -417,17 +417,17 @@ def is_finite(cls, obj: Domain) -> TypeGuard[FiniteDomain]:
     def __getitem__(self, index: int) -> tuple[Dimension, _Rng]: ...
 
     @overload
-    def __getitem__(self, index: slice) -> Self:  # noqa: F811 # redefine unused
+    def __getitem__(self, index: slice) -> Self:  # redefine unused
```
Comment (egparedes) [inline src/gt4py/next/common.py:420]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/common.py [hunk 18]
```diff
@@ -417,17 +417,17 @@ def is_finite(cls, obj: Domain) -> TypeGuard[FiniteDomain]:
...
         ...
 
     @overload
-    def __getitem__(  # noqa: F811 # redefine unused
+    def __getitem__(  # redefine unused
```
Comment (egparedes) [inline src/gt4py/next/common.py:424]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/common.py [hunk 19]
```diff
@@ -417,17 +417,17 @@ def is_finite(cls, obj: Domain) -> TypeGuard[FiniteDomain]:
...
         self, index: Dimension
     ) -> tuple[Dimension, _Rng]: ...
 
-    def __getitem__(  # noqa: F811 # redefine unused
+    def __getitem__(  # redefine unused
```
Comment (egparedes) [inline src/gt4py/next/common.py:428]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/common.py [hunk 20]
```diff
@@ -417,17 +417,17 @@ def is_finite(cls, obj: Domain) -> TypeGuard[FiniteDomain]:
...
         self, index: int | slice | Dimension
-    ) -> NamedRange | Domain:  # noqa: F811 # redefine unused
+    ) -> NamedRange | Domain:  # redefine unused
```
Comment (egparedes) [inline src/gt4py/next/common.py:430]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/embedded/operators.py [hunk 21]
```diff
@@ -42,17 +42,21 @@ class ScanOperator(EmbeddedOperator[_R, _P]):
     init: core_defs.Scalar | tuple[core_defs.Scalar | tuple, ...]
     axis: common.Dimension
 
-    def __call__(self, *args: common.Field | core_defs.Scalar, **kwargs: common.Field | core_defs.Scalar) -> common.Field:  # type: ignore[override] # we cannot properly type annotate relative to self.fun
+    def __call__(  # type: ignore[override]
+        self,
+        *args: common.Field | core_defs.Scalar,
+        **kwargs: common.Field | core_defs.Scalar,  # type: ignore[override]
+    ) -> common.Field:  # we cannot properly type annotate relative to self.fun
```
Comment (egparedes) [inline src/gt4py/next/embedded/operators.py:49]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/ffront/field_operator_ast.py [hunk 22]
```diff
@@ -51,8 +51,8 @@ def __str__(self):
 #       class Symbol(eve.GenericNode, LocatedNode, Generic[SymbolT]):
 #
 class Symbol(LocatedNode, Generic[SymbolT]):
-    id: Coerced[SymbolName]  # noqa: A003  # shadowing a python builtin
-    type: Union[SymbolT, ts.DeferredType]  # noqa A003
+    id: Coerced[SymbolName]  # shadowing a python builtin
+    type: Union[SymbolT, ts.DeferredType]  # A003
```
Comment (egparedes) [inline src/gt4py/next/ffront/field_operator_ast.py:55]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/ffront/field_operator_ast.py [hunk 23]
```diff
@@ -75,11 +75,11 @@ class Symbol(LocatedNode, Generic[SymbolT]):
 
 
 class Expr(LocatedNode):
-    type: ts.TypeSpec = ts.DeferredType(constraint=None)  # noqa A003
+    type: ts.TypeSpec = ts.DeferredType(constraint=None)  # A003
 
 
 class Name(Expr):
-    id: Coerced[SymbolRef]  # noqa: A003  # shadowing a python builtin
+    id: Coerced[SymbolRef]  # shadowing a python builtin
```
Comment (egparedes) [inline src/gt4py/next/ffront/field_operator_ast.py:82]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/ffront/field_operator_ast.py [hunk 24]
```diff
@@ -157,7 +157,7 @@ class Stmt(LocatedNode): ...
 
 
 class Starred(Expr):
-    id: Union[FieldSymbol, TupleSymbol, ScalarSymbol]  # noqa: A003  # shadowing a python builtin
+    id: Union[FieldSymbol, TupleSymbol, ScalarSymbol]  # shadowing a python builtin
```
Comment (egparedes) [inline src/gt4py/next/ffront/field_operator_ast.py:160]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/ffront/field_operator_ast.py [hunk 25]
```diff
@@ -198,29 +198,27 @@ def _collect_common_symbols(cls: type[IfStmt], instance: IfStmt) -> None:
 
 
 class FunctionDefinition(LocatedNode, SymbolTableTrait):
-    id: Coerced[SymbolName]  # noqa: A003  # shadowing a python builtin
+    id: Coerced[SymbolName]  # shadowing a python builtin
```
Comment (egparedes) [inline src/gt4py/next/ffront/field_operator_ast.py:201]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/ffront/field_operator_ast.py [hunk 26]
```diff
@@ -198,29 +198,27 @@ def _collect_common_symbols(cls: type[IfStmt], instance: IfStmt) -> None:
...
     closure_vars: list[Symbol]
-    type: Union[ts.FunctionType, ts.DeferredType] = ts.DeferredType(  # noqa: A003
-        constraint=ts.FunctionType
-    )
+    type: Union[ts.FunctionType, ts.DeferredType] = ts.DeferredType(constraint=ts.FunctionType)
 
 
 class FieldOperator(LocatedNode, SymbolTableTrait):
-    id: Coerced[SymbolName]  # noqa: A003  # shadowing a python builtin
+    id: Coerced[SymbolName]  # shadowing a python builtin
```
Comment (egparedes) [inline src/gt4py/next/ffront/field_operator_ast.py:209]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/ffront/field_operator_ast.py [hunk 27]
```diff
@@ -198,29 +198,27 @@ def _collect_common_symbols(cls: type[IfStmt], instance: IfStmt) -> None:
...
     definition: FunctionDefinition
-    type: Union[ts_ffront.FieldOperatorType, ts.DeferredType] = ts.DeferredType(  # noqa: A003
+    type: Union[ts_ffront.FieldOperatorType, ts.DeferredType] = ts.DeferredType(
```
Comment (egparedes) [inline src/gt4py/next/ffront/field_operator_ast.py:211]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/ffront/field_operator_ast.py [hunk 28]
```diff
@@ -198,29 +198,27 @@ def _collect_common_symbols(cls: type[IfStmt], instance: IfStmt) -> None:
...
         constraint=ts_ffront.FieldOperatorType
     )
 
 
 class ScanOperator(LocatedNode, SymbolTableTrait):
-    id: Coerced[SymbolName]  # noqa: A003 # shadowing a python builtin
+    id: Coerced[SymbolName]  # shadowing a python builtin
```
Comment (egparedes) [inline src/gt4py/next/ffront/field_operator_ast.py:217]: Remove unneeded comment for previously existing `#noqa` code

File: src/gt4py/next/ffront/field_operator_ast.py [hunk 29]
```diff
@@ -198,29 +198,27 @@ def _collect_common_symbols(cls: type[IfStmt], instance: IfStmt) -> None:
...
     axis: Constant
     forward: Constant
     init: Constant
     definition: FunctionDefinition  # scan pass
-    type: Union[ts_ffront.ScanOperatorType, ts.DeferredType] = ts.DeferredType(  # noqa: A003
+    type: Union[ts_ffront.ScanOperatorType, ts.DeferredType] = ts.DeferredType(
```
Comment (egparedes) [inline src/gt4py/next/ffront/field_operator_ast.py:222]: Remove unneeded comment for previously existing `#noqa` code

**Review by egparedes — CHANGES_REQUESTED** [review]
Here is the first round of feedback. Didn't have time to review all the changes yet but I'll do in the second pass, once this comments and the merge conflicts are fixed.

File: .pre-commit-config.yaml [hunk 30]
```diff
@@ -206,7 +103,7 @@ repos:
     # Add all type stubs from typeshed
     - types-all
     args: [--no-install-types]
-    exclude: |
+    exclude: |-
```
[the remark this answers is not in the document]
Comment (nfarabullini) [reply .pre-commit-config.yaml:106]: no idea

File: src/gt4py/next/program_processors/runners/dace_iterator/__init__.py [hunk 31]
```diff
@@ -304,9 +304,7 @@ def build_sdfg_from_itir(
                     f"{nested_sdfg.label} does not have debuginfo. Consider adding them in the corresponding nested sdfg.",
                     stacklevel=2,
                 ),
-                getframeinfo(
-                    currentframe()  # type: ignore
-                ),
+                getframeinfo(currentframe()),  # type: ignore
```
Comment (nfarabullini) [inline src/gt4py/next/program_processors/runners/dace_iterator/__init__.py:307]: can you add here what type ignore it is?

**Review by egparedes — APPROVED** [review]
It looks great. Thanks for all the work!

**Pull request merged**